cmd/konvoy-image/cmd/args.go

```diff
@@ -16,3 +16,7 @@ func addClusterArgs(fs *pflag.FlagSet, kubernetesVersion, containerdVersion *str
 	fs.StringVar(kubernetesVersion, "kubernetes-version", "", "The version of kubernetes to install. Example: 1.21.6")
 	fs.StringVar(containerdVersion, "containerd-version", "", "the version of containerd to install")
 }
+
+func addWorkDirArg(fs *pflag.FlagSet, workDir *string) {
+	fs.StringVar(workDir, "work-dir", "", "path to custom work directory generated by the command")
+}
```

Comment:
Im not understanding what this flag is for, somehow related to the `generate` command?
```
➜  konvoy-image-bundle-v1.19.7_darwin_amd64 ./konvoy-image generate aws --help
generate files relating to building aws images

Usage:
  konvoy-image generate aws <image.yaml> [flags]

Examples:
aws --region us-west-2 --source-ami=ami-12345abcdef images/ami/centos-79.yaml

Flags:
      --ami-groups stringArray           a list of AWS groups which are allowed use the image, using 'all' result in a public image
      --ami-regions stringArray          a list of regions to publish amis
      --ami-users stringArray            a list AWS user accounts which are allowed use the image
      --containerd-version string        the version of containerd to install
      --extra-vars strings               flag passed Ansible's extra-vars
  -h, --help                             help for aws
      --instance-type string             instance type used to build the AMI; the type must be present in the region in which the AMI is built
      --kubernetes-version string        The version of kubernetes to install. Example: 1.21.6
      --overrides strings                a comma separated list of override YAML files
      --region string                    the region in which to build the AMI
      --source-ami string                the ID of the AMI to use as the source; must be present in the region in which the AMI is built
      --source-ami-filter-name string    restricts the set of source AMIs to ones whose Name matches filter
      --source-ami-filter-owner string   restricts the source AMI to ones with this owner ID

Global Flags:
      --color     enable color output (default true)
  -v, --v int     select verbosity level, should be between 0 and 6
      --verbose   enable debug level logging (same as --v 5)
```

Reply:
I see it also in `build`
```
➜  konvoy-image-bundle-v1.19.7_darwin_amd64 ./konvoy-image build aws images/ami/centos-79.yaml --work-dir dkoshkin
INFO[0000] using workDir provided by --work-dir flag: dkoshkin
ERRO error during run: open dkoshkin/ansible_vars.yaml: no such file or directory
error encountered: exit status 3
```

Reply:
kib creates a directory where it puts all of the files it needs to run ansible and packer commands in `work-dir`. Here's how it is used in [build](https://github.com/mesosphere/konvoy-image-builder/blob/e44cceb9b97b95778fdfb58f96110405f0ce2883/pkg/app/build.go#L103)


 We're using it here because I'm using a file for extra-vars. This file allows us to handle nested maps a lot easier.

Reply:
normally this is `$(pwd)/work/<build-name>`